pennylane/templates/subroutines/qdrift.py

```diff
@@ -0,0 +1,248 @@
+
+
+        >>> args = qnp.array([1.23])
+        >>> print(qml.grad(my_circ)(*tuple(args)))
+        0.27980654844422853
+    """
+
+    def __init__(self, hamiltonian, time, n=1, seed=None, decomposition=None, id=None):
+        r"""Initialize the QDrift class"""
+
+        if isinstance(hamiltonian, Hamiltonian):
+            coeffs, ops = hamiltonian.terms()
+
+        elif isinstance(hamiltonian, Sum):
+            coeffs, ops = [], []
+            for op in hamiltonian:
+                try:
+                    coeffs.append(op.scalar)
+                    ops.append(op.base)
+                except AttributeError:  # coefficient is 1.0
+                    coeffs.append(1.0)
+                    ops.append(op)
+
+        else:
+            raise TypeError(
+                f"The given operator must be a PennyLane ~.Hamiltonian or ~.Sum got {hamiltonian}"
+            )
+
+        if any(requires_grad(coeff) for coeff in coeffs):
+            raise qml.QuantumFunctionError(
+                "The QDrift template currently doesn't support differentiation through the "
+                "coefficients of the input Hamiltonian. Please instantiate the operation "
+                "using coefficents with `requires_grad` set to False."
+            )
+
+        if decomposition is None:  # need to do this to allow flatten and _unflatten
+            unwrapped_coeffs = unwrap(coeffs)
+            decomposition = _sample_decomposition(unwrapped_coeffs, ops, time, n=n, seed=seed)
+
+        self._hyperparameters = {
+            "n": n,
+            "seed": seed,
+            "base": hamiltonian,
+            "decomposition": decomposition,
+        }
+        super().__init__(time, wires=hamiltonian.wires, id=id)
+
+    @classmethod
+    def _unflatten(cls, data, metadata):
+        """Recreate an operation from its serialized format.
+
+        Args:
+            data: the trainable component of the operation
+            metadata: the non-trainable component of the operation.
+
+        The output of ``Operator._flatten`` and the class type must be sufficient to reconstruct the original
+        operation with ``Operator._unflatten``.
+
+        **Example:**
+
+        >>> op = qml.Rot(1.2, 2.3, 3.4, wires=0)
+        >>> op._flatten()
+        ((1.2, 2.3, 3.4), (<Wires = [0]>, ()))
+        >>> qml.Rot._unflatten(*op._flatten())
+        >>> op = qml.PauliRot(1.2, "XY", wires=(0,1))
+        >>> op._flatten()
+        ((1.2,), (<Wires = [0, 1]>, (('pauli_word', 'XY'),)))
+        >>> op = qml.ctrl(qml.U2(3.4, 4.5, wires="a"), ("b", "c") )
+        >>> type(op)._unflatten(*op._flatten())
+        Controlled(U2(3.4, 4.5, wires=['a']), control_wires=['b', 'c'])
+
+        """
+        hyperparameters_dict = dict(metadata[1])
+        hamiltonian = hyperparameters_dict.pop("base")
+        return cls(hamiltonian, *data, **hyperparameters_dict)
+
+    @staticmethod
+    def compute_decomposition(*args, **kwargs):
```

Comment:
Can we remove `*args, **kwargs` here?

Reply:
It causes another error since the syntax doesn't match the base operator `compute_decomposition` method.